lib/miq_expression.rb

```diff
@@ -1229,8 +1252,6 @@ def fields(expression = exp)
     end
   end
 
-  private
-
```

Comment:
Although it didn't *actually* make the `ruby_for_date_compare` method private, I believe the intention is for the method to be private.  Thus, I think that you need to add a `private_class_method :ruby_for_date_compare`